core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2373,6 +2373,7 @@ class SparkContext(config: SparkConf) extends Logging {
  * various Spark features.
  */
 object SparkContext extends Logging {
+  URL.setURLStreamHandlerFactory(new FsUrlStreamHandlerFactory())
```

Comment:
`SparkContext` might not be the good place. `object SharedState` might be better.

In addition, you also need to add comments to explain why you did it before this line.